stress/src/geneva_exporter.rs

```diff
@@ -0,0 +1,339 @@
+        Ok((client, None))
+    } else if let Ok(mock_endpoint) = std::env::var("MOCK_SERVER_URL") {
+        println!("Using standalone mock server at: {}", mock_endpoint);
+
+        let config = GenevaClientConfig {
+            endpoint: mock_endpoint,
+            environment: "test".to_string(),
+            account: "test".to_string(),
+            namespace: "test".to_string(),
+            region: "test".to_string(),
+            config_major_version: 1,
+            auth_method: AuthMethod::MockAuth,
+            tenant: "test".to_string(),
+            role_name: "test".to_string(),
+            role_instance: "test".to_string(),
+        };
+
+        let client = GenevaClient::new(config).await?;
+        Ok((client, None))
+    } else {
+        println!("Using wiremock Geneva endpoints");
+
+        // Setup mock server
+        let mock_server = MockServer::start().await;
+        let ingestion_endpoint = format!("{}/ingestion", mock_server.uri());
+        let (auth_token, auth_token_expiry) =
+            generate_mock_jwt_and_expiry(&ingestion_endpoint, 24 * 3600);
+
+        // Mock config service (GET)
+        Mock::given(method("GET"))
+            .respond_with(ResponseTemplate::new(200).set_body_string(format!(
+                r#"{{
+                    "IngestionGatewayInfo": {{
+                        "Endpoint": "{}/ingestion",
+                        "AuthToken": "{auth_token}",
+                        "AuthTokenExpiryTime": "{auth_token_expiry}"
+                    }},
+                    "StorageAccountKeys": [{{
+                        "AccountMonikerName": "testdiagaccount",
+                        "AccountGroupName": "testgroup",
+                        "IsPrimaryMoniker": true
+                    }}],
+                    "TagId": "test"
+                }}"#,
+                mock_server.uri()
+            )))
+            .mount(&mock_server)
+            .await;
+
+        // Mock ingestion service (POST)
+        Mock::given(method("POST"))
+            .and(path_regex(r"^/ingestion.*"))
+            .respond_with({
+                ResponseTemplate::new(202).set_body_string(r#"{"ticket": "accepted"}"#)
+            })
+            .mount(&mock_server)
+            .await;
+
+        let config = GenevaClientConfig {
+            endpoint: mock_server.uri(),
+            environment: "test".to_string(),
+            account: "test".to_string(),
+            namespace: "test".to_string(),
+            region: "test".to_string(),
+            config_major_version: 1,
+            auth_method: AuthMethod::MockAuth,
+            tenant: "test".to_string(),
+            role_name: "test".to_string(),
+            role_instance: "test".to_string(),
+        };
+
+        let client = GenevaClient::new(config).await?;
+        Ok((client, Some(mock_server.uri())))
+    }
+}
+
+#[tokio::main(flavor = "current_thread")]
+//#[tokio::main(flavor = "multi_thread")]
```

Comment:
It would be good to also make the runtime type a command line argument.

Reply:
Good point. done.